fuel-tx/src/transaction/types/blob.rs

```diff
@@ -0,0 +1,235 @@
+use crate::{
+    transaction::{
+        fee::min_gas,
+        id::PrepareSign,
+        metadata::CommonMetadata,
+        types::chargeable_transaction::{
+            ChargeableMetadata,
+            ChargeableTransaction,
+            UniqueFormatValidityChecks,
+        },
+        Chargeable,
+    },
+    ConsensusParameters,
+    FeeParameters,
+    GasCosts,
+    Input,
+    Output,
+    TransactionRepr,
+    ValidityError,
+};
+use derivative::Derivative;
+use fuel_types::{
+    bytes::WORD_SIZE,
+    canonical::Serialize,
+    BlobId,
+    ChainId,
+    Word,
+};
+
+/// Adds method to `BlobId` to compute the it from blob data.
+pub trait BlobIdExt {
+    /// Computes the `BlobId` from by hashing the given data.
+    fn compute(data: &[u8]) -> Self;
+}
+
+impl BlobIdExt for BlobId {
+    fn compute(data: &[u8]) -> Self {
+        Self::new(*fuel_crypto::Hasher::hash(data))
+    }
+}
+
+pub type Blob = ChargeableTransaction<BlobBody, BlobMetadata>;
+
+#[derive(Default, Debug, Clone, PartialEq, Eq, Hash)]
+pub struct BlobMetadata;
```

Comment:
I think it would be nice to cache the hash of the blob

Reply:
I don't think we can ever use the cached value for anything. The only use of the hash is to be a nonce.